### PR TITLE
feat/enable anonymous metrics in all branches

### PR DESCRIPTION
**Pull request by aarmoa**
**Before submitting this PR, please make sure**:

- [x] Your code builds clean without any errors or warnings
- [x] You are using approved title ("feat/", "fix/", "docs/", "refactor/")

**A description of the changes proposed in the pull request**:
Refactor of the logic that collects anonymous trade volume metrics and sends the information to Hummingbot Foundation. The previous logic that associated metrics collection to processing the logs has been removed. The new logic associates the trades volume metrics collection directly to the connectors, and used the events to gather trades (or order fills).
The configuration value `heartbeat_enabled` was renamed to `anonymized_metrics_enabled`.
The configuration value 'heartbeat_interval_min` was renamed to `anonymized_metrics_interval_min`.

**Tests performed by the developer**:
Added new unit tests.
Executed a Pure Market Making strategy connected to Binance spot. I validated that the metrics appeared correctly in Datadog dashboard for metric `filled_usdt_volume`

**Tips for QA testing**:
Check that metrics collection is active (`anonymized_metrics_enabled` should be configured with value True). The collection frequency can be increased by reducing the value of `anonymized_metrics_interval_min` (set it to 0.2 for example). You will need the value of the configuration `instance_id` to filter results in Datadog dashboard.

NOTE: The PR is based on CoinAlpha:feat/allow_fees_in_different_tokens to have the RateOracle initialized and running when connectors are started. This is required when the strategy running is trading a pair with quote token different than USDT

[ch-23464]
[ch-23683]

### CODE REVIEW

**Comment by PtrckM** [comment]
On going test...

**Comment by PtrckM** [comment]
### Testing update 🚀 :

Tested the following and confirmed changes are working as expected:
- ascendex ✅ 
- binance / binance perpetual / binance_us ✅ 
- kucoin ✅ 
- gate_io ✅
- huobi ✅ 
- liquid ✅ 
- okex ✅ 
- kraken ✅ 
- coinbase_pro ✅
- terra ✅ 
- crypto_com ✅

Strategy tested so far: 
- pure_mm ✅ 
- avellaneda ✅ 
- liquidity_mining strategy ✅ 
- amm_arb ✅ 
- xemm ✅ 
- perp_mm ✅ 
- spot_perp_arb ✅ 
- twap ✅ 
- hedge ✅ 

Build: source and docker ✅ 
Tested on both USDT and non-USDT quote ✅ 

Rate oracle source:
 - binance ✅ 
 - kucoin ✅ 
 - ascendex ✅ 
 - coingecko ✅ 

---


## Ongoing test for other connectors and strategy... 👷

**Comment by aarmoa** [comment]
That is awesome @PtrckM, thank you for your effort

File: test/hummingbot/connector/test_connector_metrics_collector.py [hunk 1]
```diff
@@ -0,0 +1,289 @@
...
+            instance_id=self.instance_id,
+            client_version=self.client_version)
+
+    def tearDown(self) -> None:
+        hummingbot.connector.connector_metrics_collector.CLIENT_VERSION = self.original_client_version
+        self._reset_global_config()
+        super().tearDown()
+
+    def _reset_global_config(self):
+        for key, value in self._global_config_backup.items():
```
Comment (petioptrv) [inline test/hummingbot/connector/test_connector_metrics_collector.py:53]: I just realized that this should prob start with `global_config_map.clear()` to remove any new key-values added

**Review by petioptrv — CHANGES_REQUESTED** [review]
I've suggested two small changes

**Comment by RC-13** [comment]
Testing update:  tested against USDT markets, pending partial fills and non-USDT markets.
    * Bitfinex: Full fill = Ok, Partial fill= Ok
    * BitMart: Full fill = Ok
    * Bittrex: Full fill = Ok
    * Blocktane: Full fill = Ok
    * Coinzoom: Full fill = Ok
    * Digifinex: Full fill = Ok
    * Dydx_perpetual: Full fill = Ok
    * FTX: Full fill = Ok
    * HitBTC: Full fill = Ok
    * MEXC: Full fill = Ok
    * NDAX: Full fill = Ok
    * Probit: Full fill = Ok,
    * WazirX: Full fill = Ok

File: hummingbot/connector/connector_metrics_collector.py [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+                trade_base, trade_quote = split_hb_trading_pair(fill_event.trading_pair)
+                rate = self._rate_provider.rate(combine_to_hb_trading_pair(base=trade_quote, quote=self._valuation_token))
+
+                if rate is not None:
+                    total_volume += fill_event.amount * fill_event.price * rate
+                else:
+                    rate = self._rate_provider.rate(
+                        combine_to_hb_trading_pair(base=trade_base, quote=self._valuation_token))
+                    if rate is not None:
+                        total_volume += fill_event.amount * rate
```
[the remark this answers is not in the document]
Comment (aarmoa) [reply hummingbot/connector/connector_metrics_collector.py:160]: I am not sure about that @petioptrv. The anonymous metrics collection is not a key functionality in the client. In fact it is nothing of value to the user really, and as such I don't believe they should receive any notification from the collector's actions.

**Review by petioptrv — APPROVED** [review]
The changes LGTM!

**Review by PtrckM — APPROVED** [review]
Tested and approved by QA team.
Test summary:

- Created a pure_mm strategy and went through initial prompts successfully
- Confirmed wallet and exchange balances are reflecting correct
- Started the strategy without issues, confirmed orders are created and cancelled based on config
- Ran a test bot for hours without errors
- Confirmed in logs that there are no errors when orders are filled
- Trade information are correct (price, size) based on history command output
- Compared trades in client and exchange matched.
- Confirm metrics are being captured and converted to USD
- Confirmed metric interval is working as expected.
- Confirm rate_oracle are working as expected.
- Repeat steps and apply different strategies and connectors that are supported, and confirmed working as well.
- Confirmed working on USDT and non-USDT assets.
- Confirmed working on partial and full filled orders.

**Pull request closed without merging**